bin/sirius-script-procedures-machine-shutdown.py

```diff
@@ -0,0 +1,1097 @@
+        self.log('Step 10: Turning off modulators...')
+
+        devs = (self._devrefs['limod1'], self._devrefs['limod2'])
+
+        self.log('Disabling CHARGE...')
+        self._set_devices_propty(devs, 'CHARGE', 0)
+        if not self._wait_devices_propty(devs, 'CHARGE', 0):
+            self.log('ERR:Could not disable Modulators CHARGE.')
+            return False
+
+        _time.sleep(1.0)
+
+        self.log('...done. Disabling TRIGOUT...')
+        self._set_devices_propty(devs, 'TRIGOUT', 0)
+        if not self._wait_devices_propty(devs, 'TRIGOUT', 0):
+            self.log('ERR:Could not disable Modulators TRIGOUT.')
+            return False
+
+        self.log('...done.')
+        return True
+
+    def s11_adjust_egunbias(self):
+        """Adjust Bias Voltage to -100V."""
+        self.log('Step 11: Adjusting EGun bias voltage...')
+
+        injctrl = self._devrefs['injctrl']
+
+        self.log('Setting bias voltage to -100V...')
+        if injctrl.injtype_str == 'SingleBunch':
+            injctrl.bias_volt_sglbun = -100.0
+        else:
+            injctrl.bias_volt_multbun = -100.0
+        injctrl.wait_bias_volt_cmd_finish(timeout=10)
+        return True
+
+    def s12_adjust_egunfilament(self):
+        """Adjust EGun Bias Filament current."""
+        self.log('Step 12: Adjusting EGun Bias Filament current...')
+
+        injctrl = self._devrefs['injctrl']
+        egfila = self._devrefs['egfila']
+
+        self.log('Setting Bias Filament current to 1.1A...')
+        injctrl.filacurr_opvalue = 1.1
+        injctrl.wait_filacurr_cmd_finish(timeout=10)
+        if not egfila.wait_current(1.1, timeout=1):
+            self.log('WARN:Could not adjust EGun filament.')
+            self.log('WARN:Continuing anyway...')
+        else:
+            self.log('...done.')
+        return True
+
+    def s13_disable_egun_highvoltage(self):
+        """Disable egun high voltage."""
+        self.log('Step 13: Disabling EGun high voltage...')
+
+        injctrl = self._devrefs['injctrl']
+        eghv = self._devrefs['eghvolt']
+
+        self.log('Setting EGun High Voltage to 0V...')
+        injctrl.hvolt_opvalue = 0.0
+        injctrl.wait_hvolt_cmd_finish(timeout=100)
+        if not eghv.wait_voltage(0, timeout=1):
+            self.log('ERR:Timed out waiting for egun high voltage.')
+            return False
+
+        if not self.continue_execution():
+            return False
+
+        self.log('...done. Disabling EGun High Voltage Enable State...')
+        if not eghv.cmd_turn_off(timeout=5):
+            self.log('ERR:Could not disable egun high voltage.')
+            return False
+
+        self.log('...done.')
+        return True
+
+    def s14_start_counter(self):
+        """Check whether the countdown to tunnel access has started."""
+        self.log('Step 14: Checking tunnel access countdown...')
+
+        msg = (
+            'Check whether the countdown to tunnel access has started.')
+        input(msg)
```

Comment:
Are you sure we want this interactive behavior? As far as I understood, this script should be configured to run automatically in a pre-determined time.

Reply:
I think we could at least put a timeout waiting for the response and continue otherwise.

Reply:
Hi @fernandohds564 ! This is only for the first tests. We also have to define which steps will block the procedure.

Reply:
but we could at least put a timeout here, couldn't we?
In case you decide for the timeout, the following code works:
```python3
import select
import sys

def input_(msg, timeout=5):
    print(msg, end='')
    sys.stdout.flush()
    i, o, e = select.select([sys.stdin], [], [], 5)
    if i:
        return sys.stdin.readline().strip()
    else:
        print('timeout')
    return

out = input_('Type anything and press Enter (you have 5s to do so): ', 5) 
```

Reply:
We will replace this with the following commented lines, that checks the new PV with the remaining time for tunnel access, as soon as the interlock IOC is ready

Reply:
...done!